collectors/PropertiesCollector.py

```diff
@@ -13,6 +13,9 @@ def get_labels(self, resource_id: str, project_ids: list):
         raise NotImplementedError("Please Implement this method")
 
     def collect(self):
+        if self.collect_running and self.am_i_killed:
```

Comment:
`self.collect_running` is always `false` at this point, isn't it? So `self.am_i_killed` should be enough to bail out early. 
```suggestion
        if self.am_i_killed:
```

Reply:
That's correct, thanks.